### DISCUSSION TITLE
hd.cached & read dependencies

### DISCUSSION THREAD

**Discussion by tkriplean** [hyperdiv/hyperdiv#47] [Help]
The documentation for [hd.cached](https://docs.hyperdiv.io/reference/components/cached) says:

> Hyperdiv functions that generate UI components can be wrapped in [@]cached to avoid re-running those function calls if their read dependencies have not changed.

This is the only place I could find where the term "read dependency" is mentioned. Can you say more about what does or does not create a read dependency? After reading the short description, I feel like I always want to be caching. Are there common situations where we would NOT want to cache a view (e.g. sqlite data changing, provided to a function via a task)? 

My reference point here is Statebus's fetch(key), which creates a sort of read dependency on the state stored at key for the function calling the fetch, and those functions will be re-run when the respective state changes. I'm trying to understand the similarities and differences between the two frameworks for my mental model of what is happening under the hood.

Thanks!

**Comment by mariusnita** [comment]
A read dependency is a prop read, like `button.clicked`, `checkbox.checked`, or `my_state.count`. You may always want to be caching, though I wouldn't worry about it in the early stages of building.

Hyperdiv fundamentally works by recording which props were read, and re-running the app when one of those read props changed. `@cached` is the fundamental mechanism that implements this, and the main function is made reactive by being automatically wrapped in `@cached`.
